pkg/helmreconciler/listeners.go

```diff
@@ -0,0 +1,472 @@
+package helmreconciler
+
+import (
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/helm/pkg/manifest"
+
+	"istio.io/operator/pkg/controller/common"
+)
+
+// CompositeRenderingListener is an implementation of RenderingListener which is composed of an array of listeners.
+// All methods are delegated to each element in the Listeners array.  For completion events (e.g. EndResource()), the
+// delegates are invoked last to first.
+type CompositeRenderingListener struct {
+	// Listeners represents a list of Listeners to which this object will delegate calls.
+	Listeners []RenderingListener
+}
+
+var _ RenderingListener = &CompositeRenderingListener{}
+var _ ReconcilerListener = &CompositeRenderingListener{}
```

Comment:
remove

Reply:
Those provide type checking, ensuring that CompositeRenderingListener properly implements those interfaces.

Reply:
If it doesn't implement it then you won't be able to pass it through an interface during compile time. If you don't pass it through an interface, then you don't need the interface definition (YAGNI). 
I looked through the k8s codebase and the only reason I can see for having these is where an implementation is provided/generated which is not used in the same compilation unit. That's not the case here so all this is doing is adding dead code and making things less readable. 
But I may be misunderstanding/misreading so I'm totally open to soliciting outside opinions on this.

Reply:
Please don't resolve conversations that are not resolved. If we agree to disagree that's one thing and the path forward for that is for a TL to step in to decide.